sdk/state/ingest.go

```diff
@@ -15,71 +15,67 @@ import (
 // determine if successful or not, and understand changes in the ledger as a
 // result.
 func (c *Channel) IngestTx(tx *txnbuild.Transaction) error {
-	// Use the tx to update the latest unauthorized close agreement if possible.
-	err := c.ingestTxToUpdateUnauthorizedCloseAgreement(tx)
+	// TODO: Use the transaction result to affect on success/failure.
+
+	err := c.ingestTxToUpdateCloseState(tx)
 	if err != nil {
 		return err
 	}
 
-	// TODO: If the tx hash matches an authorized or unauthorized declaration,
-	// mark the channel as closing.
-	// TODO: If the tx hash matches an authorized or unauthorized close, mark the
-	// channel as closed.
-	// TODO: If the tx is for an older declaration, mark the channel as closing with
-	// requiring bump.
-	// TODO: Use the transaction result to affect on success/failure.
+	return nil
+}
+
+func (c *Channel) ingestTxToUpdateCloseState(tx *txnbuild.Transaction) error {
+	// If the transaction's source account is not the initiator's escrow
+	// account, then the transaction is not a part of a close agreement.
+	if tx.SourceAccount().AccountID != c.initiatorEscrowAccount().Address.Address() {
+		return fmt.Errorf(`transaction source account is not the initiator escrow account,
+			found: %s, should be: %s`, tx.SourceAccount().AccountID, c.initiatorEscrowAccount().Address.Address())
```

Comment:
I don't think we want an error in this case. In the current logic the `IngestTx` function will be a no-op with no error if you pass in a tx that has no affect on the channel state. This PR changes that to error when a tx doesn't affect the state.

The ingest function will eventually need to handle tx's from multiple accounts, for handling things like withdrawals, where any account can be the source.

Reply:
👍 
https://github.com/stellar/experimental-payment-channels/pull/232/commits/bc6c6bc6c1ed02e3b54f7fc2f4a29a3c72d22f5b